src/app/node-data/component.ts

```diff
@@ -156,9 +162,44 @@ export class NodeDataComponent extends BaseFormValidator implements OnInit, OnDe
         }
       });
 
+    // Cross validate autoscaling options.
+    merge(
+      this.form.get(Controls.MinReplicas).valueChanges,
+      this.form.get(Controls.MaxReplicas).valueChanges,
+      this.form.get(Controls.UseAutoscaling).valueChanges
+    )
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(_ => {
+        const useAutoscaling = !!this.form.get(Controls.UseAutoscaling).value;
+        if (!useAutoscaling) {
+          this.form.get(Controls.MinReplicas).setErrors(null);
+          this.form.get(Controls.MaxReplicas).setErrors(null);
+          return;
+        }
+
+        const minR = this.form.get(Controls.MinReplicas).value || 0;
+        if (minR < 0) {
+          this.form.get(Controls.MinReplicas).setErrors({min: true});
+        } else {
+          this.form.get(Controls.MinReplicas).setErrors(null);
+        }
+
+        const maxR = this.form.get(Controls.MaxReplicas).value;
+        if (maxR < minR) {
+          this.form.get(Controls.MaxReplicas).setErrors({lessThanMin: true});
```

Comment:
It would be great to create a validator for that instead of doing it from this subscription.

Reply:
I need current values of min replicas and max replicas here. How do I write a validator that can access both values?

Reply:
Ok, then let's not make it too complex and leave it as it is.